packages/core/src/Synchronizer.ts

```diff
@@ -181,101 +204,144 @@ export class Synchronizer extends EventEmitter {
         }
     }
 
+    async findStartBlock() {
+        // look for the first attesterSignUp event
+        // no events could be emitted before this
+        const filter = this.unirepContract.filters.AttesterSignedUp(
+            this.attesterId
+        )
```

Comment:
I think it doesn't have to be a specific `attesterId`
it can just query `AttesterSignedUp` events and stores new attesterId in db
(and the `latestBlocknumber` can also be stored somewhere so that it won't query through whole blocks.

Or how can the synchronizer work with many attesters? can you write a test scripts for that?

and also I think the `poll` function should be tested.

Reply:
Actually we need to query this event to load the `startTimestamp` and `epochLength` anyway

Reply:
Got it! thank you for explaining.
We can merge this first and then add tests for that.
Though I prefer to test a new feature in the same PR.